platform/darwin/src/local_glyph_rasterizer.mm

```diff
@@ -45,47 +46,72 @@ CTFontDescriptor configuration option (although we'd have to override font size
-        }
     }
-
     
-    CTFontRef getFont() {
-        if (!fontFamily) {
-            return NULL;
+    bool isEnabled() { return fallbackFontNames; }
+    
+    CTFontDescriptorRef createFontDescriptor(const FontStack& fontStack) {
+        NSMutableArray *fontNames = [NSMutableArray arrayWithCapacity:fontStack.size() + fallbackFontNames.count];
+        for (auto& fontName : fontStack) {
+            if (fontName != util::LAST_RESORT_ALPHABETIC_FONT && fontName != util::LAST_RESORT_PAN_UNICODE_FONT) {
+                [fontNames addObject:@(fontName.c_str())];
+            }
         }
+        [fontNames addObjectsFromArray:fallbackFontNames];
         
-        if (!fontHandle) {
-          NSDictionary *fontAttributes = @{
-                (NSString *)kCTFontSizeAttribute: [NSNumber numberWithFloat:24.0],
-                (NSString *)kCTFontFamilyNameAttribute: [[NSString alloc] initWithCString:fontFamily->c_str() encoding:NSUTF8StringEncoding]
+        CFMutableArrayRefHandle fontDescriptors(CFArrayCreateMutable(kCFAllocatorDefault, fontNames.count, &kCFTypeArrayCallBacks));
+        for (NSString *name in fontNames) {
+            NSDictionary *fontAttributes = @{
+                (NSString *)kCTFontSizeAttribute: @(util::ONE_EM),
+                (NSString *)kCTFontNameAttribute: name,
+                (NSString *)kCTFontDisplayNameAttribute: name,
+                (NSString *)kCTFontFamilyNameAttribute: name,
             };
-
+            
             CTFontDescriptorRefHandle descriptor(CTFontDescriptorCreateWithAttributes((CFDictionaryRef)fontAttributes));
-            fontHandle = CTFontCreateWithFontDescriptor(*descriptor, 0.0, NULL);
-            if (!fontHandle) {
-                throw std::runtime_error("CTFontCreateWithFontDescriptor failed");
-            }
+            CFArrayAppendValue(*fontDescriptors, *descriptor);
+        }
+
+        CFStringRef keys[] = { kCTFontSizeAttribute,                  kCTFontCascadeListAttribute };
+        CFTypeRef values[] = { (__bridge CFNumberRef)@(util::ONE_EM), *fontDescriptors };
+
+        CFDictionaryRefHandle attributes(
+            CFDictionaryCreate(kCFAllocatorDefault, (const void**)&keys,
+                (const void**)&values, sizeof(keys) / sizeof(keys[0]),
+                &kCFTypeDictionaryKeyCallBacks,
+                &kCFTypeDictionaryValueCallBacks));
+        if (CFArrayGetCount(*fontDescriptors)) {
+            CTFontDescriptorRef firstDescriptor = (CTFontDescriptorRef)CFArrayGetValueAtIndex(*fontDescriptors, 0);
+            return CTFontDescriptorCreateCopyWithAttributes(firstDescriptor, *attributes);
+        } else {
+            return CTFontDescriptorCreateWithAttributes(*attributes);
         }
-        return fontHandle;
+    }
+
+    CTFontRef createFont(const FontStack& fontStack) {
+        CTFontDescriptorRefHandle descriptor(createFontDescriptor(fontStack));
+        return CTFontCreateWithFontDescriptor(*descriptor, 0.0, NULL);
```

Comment:
Related question for the CTFont that gets created here - where does that get destroyed?

Reply:
Because this method follows the Create rule, the one call site in `LocalGlyphRasterizer::rasterizeGlyph()` puts the return value in a `CTFontRefHandle`, which automatically releases the `CTFontRef` when it goes out of scope. I’m open to using a C++ pointer wrapper over C convention if we can do that confidently.

Reply:
I'm not sure that's necessary. Perhaps add some comments about the usage of `...RefHandle` to release objects?

Reply:
I’ll add comments about the Create rule. `CTFontDescriptorRefHandle` usage is documented in CFHandle.hpp along with important caveats.